torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -240,12 +243,17 @@ def _fill_tensor_shape_type(
             # None could be a valid value for return type, so we need to handle it.
             # e.g. the function: meta__scaled_dot_product_flash() in cpu mode.
             continue
-        elif isinstance(expected_value, (torch.SymInt, torch.SymFloat, torch.SymBool)):
+        elif fx_type_utils.is_torch_symbolic_type(expected_value):
             # aten::sym_size output is a int, not a tensor, which stands
             # for the size of one dim. We treat it as 0-D tensor.
-            onnxscript_value.dtype = fx_type_utils.from_sym_value_to_torch_dtype(
-                expected_value
-            )
+            if node.target == math.ceil or node.target == math.floor:
+                # Type promotion for ceil/floor from built-in functions.
+                # ceil and floor have float outputs in torchlib
+                onnxscript_value.dtype = torch.float32
```

Comment:
should this be part of the type promotion logic? adding hard-coded op-specific logic in the dispatcher makes it hard to maintain